### PR TITLE
Extract base event handler

### PR DESCRIPTION
**Pull request by sleistner**
This PR lays the foundation for future implementations by extracting common handler and validator functionality into a base class.

- http validator
- http handler

### CODE REVIEW

File: lambda_handlers/handlers/event_handler.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+class EventHandler(LambdaHandler, ValidationMixin, FormattingMixin):
+    """
+    Decorator class to facilitate the definition of AWS Lambda handlers with:
+        - input validation,
+        - output formatting.
+
+    Parameters
+    ----------
+    input_format: Callable
+        Formatter callable to parse the input event.
```
Comment (alexsavio) [inline lambda_handlers/handlers/event_handler.py]: Could you please mention here the defaults?

File: lambda_handlers/handlers/event_handler.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    output_format: Callable
+        Formatter callable to format the output body from the return value of the handler function.
```
Comment (alexsavio) [inline lambda_handlers/handlers/event_handler.py]: The default here as well.

File: lambda_handlers/handlers/http_handler.py [hunk 3]
```diff
@@ -37,60 +35,43 @@ class HTTPHandler(LambdaHandler):
     cors: lambda_decorator.response.CorsHeaders
         Definition of the CORS headers.
 
-    body_format: Callable
+    input_format: Callable
         Formatter callable to parse the input body.
```
Comment (alexsavio) [inline lambda_handlers/handlers/http_handler.py]: Wouldn't it be nice to mention the default here and in the output_format?

File: lambda_handlers/validators/validator.py [hunk 4]
```diff
@@ -84,7 +69,11 @@ def _validate_many(
         for key, schema in definitions.items():
             data, errors = self.validate(target.get(key, {}), schema)
             if errors:
-                cumulative_errors[key].extend(errors)
+                entry = cumulative_errors[key]
```
Comment (alexsavio) [inline lambda_handlers/validators/validator.py]: what about? :

```
                if not isinstance(errors, list):
                    errors = [errors]
                 cumulative_errors[key].extend(errors)
```

File: lambda_handlers/validators/http/tests/test_marshmallow_validator.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+        nested_errors = error.value.description
+        assert isinstance(nested_errors, list)
+        assert len(nested_errors) == 1
+
+        path_parameters_errors = next(
+            errors['pathParameters'] for errors in nested_errors if 'pathParameters' in errors
+        )
+        assert isinstance(path_parameters_errors, list)
+        # assert 'user_name' in path_parameters_errors
```
Comment (alexsavio) [inline lambda_handlers/validators/http/tests/test_marshmallow_validator.py]: would be good to remove this commented code?

File: lambda_handlers/validators/validator.py (hunk 4 above)
Comment (sleistner) [reply lambda_handlers/validators/validator.py]: done

File: lambda_handlers/validators/http/tests/test_marshmallow_validator.py (hunk 5 above)
Comment (sleistner) [reply lambda_handlers/validators/http/tests/test_marshmallow_validator.py]: done

File: lambda_handlers/handlers/event_handler.py (hunk 1 above)
Comment (sleistner) [reply lambda_handlers/handlers/event_handler.py]: done

File: lambda_handlers/handlers/event_handler.py (hunk 2 above)
Comment (sleistner) [reply lambda_handlers/handlers/event_handler.py]: done

**Comment by sleistner** [comment]
Thanks, changes pushed

**Pull request merged**